src/storage/mvcc/reader/reader.rs

```diff
@@ -192,7 +192,11 @@ impl<S: EngineSnapshot> MvccReader<S> {
                 .read()
                 .map
                 .get(key)
-                .map(|lock| lock.to_lock())
+                .map(|(lock, deleted)| {
+                    // Protected by latch, deleted must be false.
```

Comment:
I'm not very sure about whether it should read the deleted pessimistic lock. I'm actually inclined to yes.

For example, when a prewrite has marked the lock deleted but hasn't applied yet. If we use `MvccByKey` to read the key but skip the deleted lock, we will see there isn't any lock on it. It may be a strange state.

Reply:
Of course, there is also a state when the prewrite is applied but the lock isn't deleted. But this time gap is much smaller.

Reply:
> MvccByKey and MvccByStartTs can read a deleted pessimistic lock.

I don't mean these commands should read deleted locks... My point is it can trigger `debug_assert!(!deleted)`.

Reply:
I'm also thinking about checking term and version here, to solve the problem when the map is already cleared after the region becomes follower.